Fixit : Add Support for ordering and limiting results in CLI show commands

bvaradar opened the pull request:
"Show" CLI commands 
  - Allow users to sort the results by single column in CLI output
  -  Allow users to limit the results in CLI output

hoodie:schemaless_mezzanine_trips_rows->stats filesizes --partitionPath 2016/09/01 --sortBy "95th" --desc true --limit 5
    ________________________________________________________________________________________________
    | CommitTime    | Min     | 10th    | 50th    | avg     | 95th    | Max     | NumFiles| StdDev  |
    |===============================================================================================|
    | 20180419090151| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 1       | 0.0 B   |
    | 20180414172907| 106.8 MB| 106.8 MB| 106.9 MB| 106.9 MB| 107.0 MB| 107.0 MB| 2       | 124.0 KB|
    | 20180411203144| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 1       | 0.0 B   |
    | 20180416200204| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 107.0 MB| 1       | 0.0 B   |
    | 20180412150134| 106.9 MB| 106.9 MB| 106.9 MB| 106.9 MB| 106.9 MB| 106.9 MB| 2       | 34.2 KB |

bvaradar commented:
@vinothchandar @n3nash : This is a minor change as part of Fixit effort. Please take a look at it.

vinothchandar commented on hoodie-cli/src/main/java/com/uber/hoodie/cli/TableBuffer.java:
```diff
@@ -0,0 +1,177 @@
+   * Main API to add row to the buffer
+   * @param row Row
+   */
+  public TableBuffer add(List<Object> row) {
```
Can we rename without Buffer at the end

vinothchandar commented on hoodie-cli/src/main/java/com/uber/hoodie/cli/TableFieldType.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Type of fields in the table
+ */
+public enum TableFieldType {
```
Could we reuse the standard java classes like - String.class, Long.class etc - directly ?

vinothchandar replied:
This will avoid having to define indirections here

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/hudi?pullRequest=383) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/hudi?pullRequest=383) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/hudi?pullRequest=383) it.</sub>

bvaradar commented:
@vinothchandar : Addressed all the comments except the one about avoiding repeats in cli declarations. This one seems tricky as Java Annotations (used by spring-shell) is limited in flexibility (no inheritance, only static attribute values at call-site). The only way I can think of is using Annotation Processor to generate wrapper methods on the fly. I will sync up with you in person to discuss this.
